src/ast.hpp

```diff
@@ -145,90 +145,90 @@ struct Identifier : public AST {
 
 struct BinaryExpression : public AST {
 	Token const* m_op_token;
-	Own<AST> m_lhs;
-	Own<AST> m_rhs;
+	AST* m_lhs;
+	AST* m_rhs;
 
 	BinaryExpression()
 	    : AST {ASTTag::BinaryExpression} {}
 };
 
 struct CallExpression : public AST {
-	Own<AST> m_callee;
-	std::vector<Own<AST>> m_args;
+	AST* m_callee;
+	std::vector<AST*> m_args;
 
 	CallExpression()
 	    : AST {ASTTag::CallExpression} {}
 };
 
 struct IndexExpression : public AST {
-	Own<AST> m_callee;
-	Own<AST> m_index;
+	AST* m_callee;
+	AST* m_index;
 
 	IndexExpression()
 	    : AST {ASTTag::IndexExpression} {}
 };
 
 struct RecordAccessExpression : public AST {
-	Own<AST> m_record;
+	AST* m_record;
 	Token const* m_member;
 
 	RecordAccessExpression()
 	    : AST {ASTTag::RecordAccessExpression} {}
 };
 
 struct TernaryExpression : public AST {
-	Own<AST> m_condition;
-	Own<AST> m_then_expr;
-	Own<AST> m_else_expr;
+	AST* m_condition;
+	AST* m_then_expr;
+	AST* m_else_expr;
 
 	TernaryExpression()
 	    : AST {ASTTag::TernaryExpression} {}
 };
 
 struct Block : public AST {
-	std::vector<Own<AST>> m_body;
+	std::vector<AST*> m_body;
 
 	Block()
 	    : AST {ASTTag::Block} {}
 };
 
 struct ReturnStatement : public AST {
-	Own<AST> m_value;
+	AST* m_value;
 
 	ReturnStatement()
 	    : AST {ASTTag::ReturnStatement} {}
 };
 
 struct IfElseStatement : public AST {
-	Own<AST> m_condition;
-	Own<AST> m_body;
-	Own<AST> m_else_body; // can be nullptr
+	AST* m_condition;
+	AST* m_body;
+	AST* m_else_body; // can be nullptr
```

Comment:
Pointers don't get automatically initialized to nullptr, so a default value should be added.

Reply:
This is not the only place where this applies.

Reply:
Funny enough, the bug I just fixed was exactly that